app/src/components/UI/Atoms/SearchBox/SearchBox.tsx

```diff
@@ -1,26 +1,76 @@
-import React from "react";
-import { Container, Input, Menu } from "semantic-ui-react";
+import React, { ChangeEvent, SetStateAction, useState } from "react";
+import {
+  Container,
+  Input,
+  InputOnChangeData,
+  Menu,
+  Search,
+  SearchProps,
+} from "semantic-ui-react";
 import styles from "./searchbox.module.css";
 import Buttons from "../Button/Buttons";
+import { useVendorsQuery, Vendor } from "../../../../api";
+import { Field } from "formik";
 
 /**
  * This is the searchbox for the header
  */
 
+const inputBox = (
+  <Input
+    icon={
+      <Buttons enter color={"green"}>
+        Enter
+      </Buttons>
+    }
+    placeholder="Search..."
+    focus
+    size={"small"}
+    className={styles.inputBox}
+  />
+);
+
 export const SearchBox: React.FC = () => {
-  const searchEnterHandler = () => {};
+  const [searchResult, setSearchResult] = useState<Vendor[]>([]);
+  const { data: vendorsList } = useVendorsQuery();
+
+  const onSearchChange = (
+    event: React.MouseEvent<HTMLElement, MouseEvent>,
+    data: SearchProps
+  ) => {
+    if (data.value?.length === 0) {
+      setSearchResult([]);
+      return;
+    }
+
+    if (vendorsList) {
+      for (let i = 0; i < vendorsList.length; i++) {
+        if (data.value === vendorsList[i].Name) {
```

Comment:
Shouldn't a vendor be shown in the result if its name contains the string? The search suggestions are only helpful if I don't have to type the entire name.

Reply:
I can see what you mean, but just a note: the way Selina generated the fake data, the names are like "vendor81", "vendor83", "vendor85" etc. So in order to filter it down, you would have to at least type in the word "vendor".

Reply:
I understand that. In real-world usage, names will be very different.

Reply:
![image](https://user-images.githubusercontent.com/42430035/154117781-fc4a1248-6cfc-47a5-b83a-58b364c24cb3.png)
![image](https://user-images.githubusercontent.com/42430035/154117831-14d58647-c9dd-40f8-ab20-271a0e2c2386.png)

I added two different names that start with the same two letters and this is how it filters. How is that?

Reply:
I think it should match names by substring. You should be able to search between the name and not just by prefix. Lots of names can start with "The" and you shouldn't have to type "The" to search those.